pkg/costmodel/allocation.go

```diff
@@ -575,7 +604,26 @@ func (cm *CostModel) buildPodMap(window kubecost.Window, resolution, maxBatchSiz
 			return err
 		}
 
-		applyPodResults(window, resolution, podMap, clusterStart, clusterEnd, resPods)
+		// queryFmtPodsUID will return both UID-containing results, and non-UID-containing results,
+		// so filter out the non-containing results so we don't duplicate pods.
```

Comment:
Discussed this w/ Michael. This is because Kubecost has two `kube_pod_container_status_running` metrics by default (replicated and included KSM versions). Querying avg by UID will return both, hence we filter. Updated the comment to clarify this.